packages/shared/lib/core/network/utils/checkNodeUrlValidity.ts

```diff
@@ -19,6 +19,13 @@ export function checkNodeUrlValidity(
         return 'error.node.invalid'
     }
 
+    // Check if it's HTTP and localhost
+    const isHttpLocalhost = newUrl.toLowerCase().startsWith('http://localhost') || newUrl.startsWith('http://127.0.0.1')
+    if (isHttpLocalhost && !allowInsecure) {
```

Comment:
shouldn't this be like that?
```suggestion
    if (isHttpLocalhost && allowInsecure) {
```